src/Uno.UWP/Media/Playback/MediaPlayer.wasm.cs

```diff
@@ -0,0 +1,297 @@
+﻿using System;
+using Uno.Media.Playback;
+using Windows.Media.Core;
+using Uno.Extensions;
+using System.IO;
+using Uno.Foundation.Logging;
+using System.Collections.Generic;
+using Uno;
+using Uno.Helpers;
+using System.Linq;
+
+namespace Windows.Media.Playback
+{
+	public partial class MediaPlayer
+	{
+		public IHtmlMediaPlayer Player;
+		private bool _isPlayRequested;
+		private bool _isPlayerPrepared;
+		private List<Uri> _playlistItems;
+		private int _playlistIndex;
+		const string MsAppXScheme = "ms-appx";
+
+		private void Initialize(IHtmlMediaPlayer player)
+		{
+			Player = player;
+			InitializePlayer();
+		}
+
+		private void TryDisposePlayer()
+		{
+			if (Player != null)
+			{
+				_isPlayRequested = false;
+				_isPlayerPrepared = false;
+			}
+		}
+
+		#region Player Initialization
+		private void InitializePlayer()
+		{
+			Player.OnSourceFailed -= OnError;
+			Player.OnSourceLoaded -= OnPrepared;
+			Player.OnSourceEnded -= OnCompletion;
+			Player.OnTimeUpdate -= OnTimeUpdate;
+			Player.OnSourceFailed += OnError;
+			Player.OnSourceLoaded += OnPrepared;
+			Player.OnSourceEnded += OnCompletion;
+			Player.OnTimeUpdate += OnTimeUpdate;
+
+			PlaybackSession.PlaybackStateChanged -= OnStatusChanged;
+			PlaybackSession.PlaybackStateChanged += OnStatusChanged;
+		}
+
+		private void SetPlaylistItems(MediaPlaybackList playlist)
+		{
+			_playlistItems = playlist.Items
+				.Select(i => i.Source.Uri)
+				.ToList();
```

Comment:
Can the uri change after `_playlistItems` and then we have outdated `Uri`s?

Reply:
Don't think so. in order for that to happen InitializeSource() needs to be called, which means MediaElement will be refreshed.